### PR TITLE
Add job to verify transaction statistics - Closes #660

### LINKED ISSUE #660: Conflict in transaction stats and transaction list
Only 29 transaction happened in the last 6 months in the graphical view, but in the list below shows more than 100 transaction.

### PR DESCRIPTION
**Pull request by nagdahimanshu**
### What was the problem?
This PR resolves #660 

### How was it solved?
- [x] Schedule new job `verify.transaction.statistics` (every 3 hours)
- [x] Implement new method `updateTransactionStatistics` that verify stats based on tx type
- [x] Merge changes from development

### How was it tested?
Local + Jenkins

### CODE REVIEW

File: services/core/jobs/verifyTransactionStatistics.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+
```
Comment (sameersubudhi) [inline services/core/jobs/verifyTransactionStatistics.js]: Please consider moving the job to `services/core/jobs/transactionStatistics.js`.

File: services/core/jobs/transactionStatistics.js [hunk 2]
```diff
@@ -38,4 +38,16 @@ module.exports = [
...
+	{
+		name: 'verify.transaction.statistics',
+		description: 'Keep the transaction statistics up-to-date',
+		schedule: '15 */3 * * *', // Every 3 hours at 15th minute
+		controller: async () => {
+			if (config.transactionStatistics.enabled) {
+				logger.debug('Update transaction stats...');
+				await transactionStatistics
+					.updateTransactionStatistics(config.transactionStatistics.historyLengthDays);
+			}
```
Comment (sameersubudhi) [inline services/core/jobs/transactionStatistics.js:50]: Please consider wrapping all the jobs within `try...catch` and output a warning log upon error.

File: services/core/shared/core/transactionStatistics.js [hunk 3]
```diff
@@ -260,10 +261,41 @@ const init = async historyLengthDays => {
 
 const updateTodayStats = async () => fetchTransactionsForPastNDays(1, true);
 
+const updateTransactionStatistics = async historyLengthDays => {
```
Comment (sameersubudhi) [inline services/core/shared/core/transactionStatistics.js:264]: Please consider updating the method name to reflect its exact utility.

```suggestion
const verifyTransactionStatistics = async historyLengthDays => {
```
or
```suggestion
const verifyAndUpdateTransactionStatistics = async historyLengthDays => {
```
or
```suggestion
const validateTransactionStatistics = async historyLengthDays => {
```

**Review by sameersubudhi — CHANGES_REQUESTED** [review]
Kindly consider the following:

File: services/core/shared/core/transactionStatistics.js [hunk 4]
```diff
@@ -260,10 +261,41 @@ const init = async historyLengthDays => {
...
+				toTimestamp: (moment.unix(dateTo).unix()) / 1000,
+			})).meta.total;
+			const result = txCount === distributionByType[type];
+			return result;
+		},
+		{ concurrency: Object.keys(distributionByType).length },
+	);
+
+	const isStatisticsBuilt = !verifyStatistics.some(acc => acc === false);
+	if (!isStatisticsBuilt) await fetchTransactionsForPastNDays(historyLengthDays, true);
```
Comment (sameersubudhi) [inline services/core/shared/core/transactionStatistics.js:291]: Kindly consider simplifying the code to something similar to:

```suggestion
	const isStatsProperlyBuilt = verifyStatistics.every(isVerified => isVerified);
	if (!isStatsProperlyBuilt) await fetchTransactionsForPastNDays(historyLengthDays, true);
```

**Review by sameersubudhi — APPROVED** [review]
Thank you for the changes. Works very well 👍

**Review by MichalTuleja — APPROVED** [review]

**Pull request merged**